### PR TITLE
Model 3: Implement hemorrhage, hemoglobin, and anemia observers

### PR DESCRIPTION
**Pull request by mattkappel**
### Description
- *Category*: observers
- *JIRA issue*: [MIC-2850](https://jira.ihme.washington.edu/browse/MIC-2850)
- *Research reference*: [Output table](https://vivarium-research.readthedocs.io/en/latest/concept_models/vivarium_iv_iron/maternal_model/concept_model.html#simulation-output-table)

Summary of changes:
- Adds hemoglobin exposure sums
- Adds anemia state person time
- Corrects naming of existing observers and inclusion in the location-specific yaml files.

### Verification and Testing
Ran psimulate for a location, results appeared as expected.

### CODE REVIEW

File: src/vivarium_gates_iv_iron/components/observers.py [hunk 1]
```diff
@@ -546,3 +546,160 @@ def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
...
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        self.configuration = builder.configuration.metrics.hemoglobin
+        self.start_time = get_time_stamp(builder.configuration.time.start)
+        self.hemoglobin = builder.value.get_value("hemoglobin.exposure")
+        self.step_size = builder.time.step_size()
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/observers.py]: I don't see where you use clock, start_time, or step_size. If I'm correctly reading this, you don't need any of them

**Review by albrja — APPROVED** [review]

File: src/vivarium_gates_iv_iron/components/observers.py [hunk 2]
```diff
@@ -546,3 +546,160 @@ def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
...
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        self.configuration = builder.configuration.metrics.anemia
+        self.start_time = get_time_stamp(builder.configuration.time.start)
+        self.anemia_levels = builder.value.get_value("anemia_levels")
+        self.step_size = builder.time.step_size()
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/observers.py]: You can use event.step_size, so you don't need a self.step_size

File: src/vivarium_gates_iv_iron/components/observers.py [hunk 3]
```diff
@@ -546,3 +546,160 @@ def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
...
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
+        self.configuration = builder.configuration.metrics.anemia
+        self.start_time = get_time_stamp(builder.configuration.time.start)
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/observers.py]: You're never using start_time

File: src/vivarium_gates_iv_iron/components/observers.py [hunk 4]
```diff
@@ -546,3 +546,160 @@ def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
...
+    def name(self):
+        return 'anemia_observer'
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        self.clock = builder.time.clock()
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/observers.py:822]: You can use event.time.year, so you don't need a clock.

File: src/vivarium_gates_iv_iron/components/observers.py [hunk 5]
```diff
@@ -546,3 +546,160 @@ def metrics(self, index: pd.Index, metrics: Dict) -> Dict:
...
+    def on_time_step_prepare(self, event: Event):
+        pop = self.population_view.get(event.index)
+        pop['anemia_level'] = self.anemia_levels(event.index)
+
+        def get_state_person_time(pop: pd.DataFrame, config: Dict[str, bool],
+                                  state_machine: str, state: str, pregnancy_state: str,
+                                  hemorrhage_type: str, current_year: Union[str, int],
+                                  step_size: pd.Timedelta, age_bins: pd.DataFrame) -> Dict[str, float]:
+            """Custom person time getter that handles state column name assumptions"""
+            base_key = get_output_template(**config).substitute(measure=f'{state}_anemia_person_time_among_{pregnancy_state}_with_{hemorrhage_type}',
```
Comment (rmudambi) [inline src/vivarium_gates_iv_iron/components/observers.py]: Consider breaking this and line 688 up into multiple lines for ease of reading

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_gates_iv_iron/components/observers.py (hunk 4 above)
Comment (rmudambi) [reply src/vivarium_gates_iv_iron/components/observers.py:822]: Quick note, `event.time == self.clock() + event.step_size`, so if you remove `self.clock()`, you'd want to replace it with `event.time - event.step_size`. I"m not sure that makes sense upon reflection.

Comment (mattkappel) [reply src/vivarium_gates_iv_iron/components/observers.py:822]: I decided to keep the clock here. Thanks much for the other finds!

**Pull request merged**